admin_guide/cluster_management.rst

```diff
@@ -267,3 +267,162 @@ Output:
   ├───────────────┤
   │ store_id      │
   └───────────────┘
+
+Detecting locks
+---------------
+
+This query will run across all worker nodes and identify locks, how long they've been open, and the offending queries:
+
+.. code-block:: postgresql
+
+  SELECT run_command_on_workers($cmd$
+    SELECT array_agg(
+      blocked_statement || ' $ ' || cur_stmt_blocking_proc
+      || ' $ ' || cnt::text || ' $ ' || age
+    )
+    FROM (
+      SELECT blocked_activity.query    AS blocked_statement,
+             blocking_activity.query   AS cur_stmt_blocking_proc,
+             count(*)                  AS cnt,
+             age(now(), min(blocked_activity.query_start)) AS "age"
+      FROM pg_catalog.pg_locks         blocked_locks
+      JOIN pg_catalog.pg_stat_activity blocked_activity
+        ON blocked_activity.pid = blocked_locks.pid
+      JOIN pg_catalog.pg_locks         blocking_locks
+        ON blocking_locks.locktype = blocked_locks.locktype
+       AND blocking_locks.DATABASE IS NOT DISTINCT FROM blocked_locks.DATABASE
+       AND blocking_locks.relation IS NOT DISTINCT FROM blocked_locks.relation
+       AND blocking_locks.page IS NOT DISTINCT FROM blocked_locks.page
+       AND blocking_locks.tuple IS NOT DISTINCT FROM blocked_locks.tuple
+       AND blocking_locks.virtualxid IS NOT DISTINCT FROM blocked_locks.virtualxid
+       AND blocking_locks.transactionid IS NOT DISTINCT FROM blocked_locks.transactionid
+       AND blocking_locks.classid IS NOT DISTINCT FROM blocked_locks.classid
+       AND blocking_locks.objid IS NOT DISTINCT FROM blocked_locks.objid
+       AND blocking_locks.objsubid IS NOT DISTINCT FROM blocked_locks.objsubid
+       AND blocking_locks.pid != blocked_locks.pid
+      JOIN pg_catalog.pg_stat_activity blocking_activity ON blocking_activity.pid = blocking_locks.pid
+      WHERE NOT blocked_locks.GRANTED
+       AND blocking_locks.GRANTED
+      GROUP BY blocked_activity.query,
+               blocking_activity.query
+      ORDER BY 4
+    ) a
+  $cmd$);
```

Comment:
This lock-finding query is the only one I had trouble getting output for. On the coordinator I run:

```
begin;
LOCK TABLE http_request in row share mode;
```

Then in another psql session I run the lock detecting query but it says

```
┌───────────────────────────────────────────────────────────────┐
│                    run_command_on_workers                     │
├───────────────────────────────────────────────────────────────┤
│ (ec2-13-59-96-221.us-east-2.compute.amazonaws.com,5432,t,"")  │
│ (ec2-52-14-226-167.us-east-2.compute.amazonaws.com,5432,t,"") │
└───────────────────────────────────────────────────────────────┘
```
meaning that no locks are active.

Reply:
OK, I was able to get output by running two sessions on a worker, one doing a select for update and another a regular update. That causes one statement to block on another.